signac/synced_collections/backends/collection_redis.py

```diff
@@ -0,0 +1,153 @@
+# Copyright (c) 2020 The Regents of the University of Michigan
+# All rights reserved.
+# This software is licensed under the BSD 3-Clause License.
+"""Implements a Redis SyncedCollection backend."""
+import json
+
+from .. import SyncedAttrDict, SyncedCollection, SyncedList
+
+
+class RedisCollection(SyncedCollection):
```

Comment:
Many suggestions/questions/comments from the MongoDB backend also apply to this file.